app/components/PredictionMarkets/PredictionMarkets.jsx

```diff
@@ -0,0 +1,388 @@
+import React, {Component} from "react";
+import assetUtils from "common/asset_utils";
+import AssetActions from "actions/AssetActions";
+import MarketsActions from "actions/MarketsActions";
+import counterpart from "counterpart";
+import PredictionMarketsOverviewTable from "./PredictionMarketsOverviewTable";
+import PredictionMarketDetailsTable from "./PredictionMarketDetailsTable";
+import SearchInput from "../Utility/SearchInput";
+import HelpContent from "../Utility/HelpContent";
+import AddOpinionModal from "./AddOpinionModal";
+import CreateMarketModal from "./CreateMarketModal";
+import ResolveModal from "./ResolveModal";
+import {ChainStore} from "bitsharesjs";
+import {Button} from "bitshares-ui-style-guide";
+
+export default class PredictionMarkets extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            assets: [],
+            lastAssetSymbol: null,
+            markets: [],
+            currentAccountId: null,
+            searchTerm: "",
+            detailsSearchTerm: "",
+            selectedMarket: null,
+            opinions: [],
+            preselectedOpinion: "no",
+            preselectedAmount: 0,
+            isCreateMarketModalOpen: false,
+            isAddOpinionModalOpen: false,
+            isResolveModalOpen: false
+        };
+        this._updateAssetsList("A");
+    }
+
+    componentWillReceiveProps(np) {
+        if (np.currentAccount !== this.props.currentAccount) {
```

Comment:
can you add some comments please what is done for what purpose?

Reply:
Splited into several functions